File: Practica1/extra/shadertoy2love.rb
num = 0
Dir.glob('*.frag') do |file|
	text = File.read(file)
	if  (not text =~ /uniform float iGlobalTime/)
		num = num + 1
		puts "Converting: " + file
		text = "uniform vec4 iDate; \n" + text
		text = "uniform vec3 iMouse; \n" + text
		text = "uniform sampler2D iChannel0; \n" + text
		text = "uniform vec3 iChannelResolution[4]; \n" + text
		text = "uniform float iChannelTime[4]; \n" + text
		text = "uniform vec4 iLoc; \n" + text
		text = "uniform float iGlobalTime; \n" + text
		text = "uniform vec3 iResolution; \n" + text
		File.open(file, "w") {|file| file.write(text)}
	end
end
if (num > 0)
puts "All done, press any key to exit"
else
puts "no files converted, press any key to exit"
end
gets.chomp

File: Practica1/extra/shadertoy2love.pl
#!/usr/bin/perl -i.bak -ln
use strict;

	BEGIN { @ARGV = glob '*.glsl' }
	if (!/uniform float iGlobalTime/)
	{
		if ($. == 1)
		{
			print "uniform vec3 iResolution;";
			print "uniform number iGlobalTime;";
			print "uniform vec4 iLoc;";
			print "uniform number iChannelTime[4];";
			print "uniform vec3 iChannelResolution[4];";
			print "uniform sampler2D iChannel0;";
			print "uniform vec3 iMouse;";
			print "uniform vec4 iDate;";
		}
	}
	print or die "$ARGV: $!";
	close ARGV or die("$ARGV: $!") if eof;